docs/operate/reference/viam-server/debug-endpoints.md

```diff
@@ -0,0 +1,108 @@
+date: "2025-05-19"
+---
+
+The RDK (Robot Development Kit) that powers `viam-server` includes several debug endpoints that expose server internals.
+These endpoints can be helpful during development or when troubleshooting complex issues:
+
+- [`pprof`](#pprof)
+- [`graph`](#graph)
+
+## `pprof`
+
+Setting `enable_web_profile` to `true` in your machine configuration allows you to visualize runtime profiling data in the localhost browser.
+The pprof tool provides valuable insights into your application's performance, including:
+
+- CPU usage
+- Memory allocation
+- Goroutine blocking
+- Execution tracing
+
+This information can be particularly useful when:
+- Diagnosing performance bottlenecks
+- Investigating memory leaks
+- Understanding resource usage patterns
+- Troubleshooting high CPU or memory usage
+
+For more detailed information on using pprof, refer to the [Go pprof documentation](https://pkg.go.dev/net/http/pprof).
+
+### Configuration
+
+To enable the pprof endpoints, set `enable_web_profile` to `true` in your machine configuration:
+
+1. Navigate to the **CONFIGURE** tab of your machine in the Viam app
+2. Click on the **...** menu next to your machine part in the left-hand menu
+3. Select **Edit JSON**
+4. Add the `enable_web_profile` field at the root level of your configuration:
+
+```json
+{
+  "components": [...],
+  "services": [...],
+  "enable_web_profile": true
+}
+```
+
+### Available Endpoints
+
+Once enabled, you can access the pprof interface by navigating to `https://localhost:8080/debug/pprof/` in your browser.
+
+The following pprof routes are available:
+
+1. `/debug/pprof/cmdline` - Command line arguments used to start the process
+2. `/debug/pprof/profile` - CPU profile data (30-second sample by default)
+3. `/debug/pprof/symbol` - Symbol lookup for program counters
+4. `/debug/pprof/trace` - Execution trace data
+
+## `graph`
+
+The `/debug/graph` endpoint shows a graphical representation of the machine resources managed by the RDK.
+The graph is created using [graphviz](https://graphviz.org/).
+
+You can access this endpoint by navigating to `https://localhost:8080/debug/graph` in your browser when connected to a machine running `viam-server`.
+
+### Layout Options
+
+You can specify different [layouts](https://graphviz.org/docs/layouts/) by passing in the query parameter `?layout=LAYOUT`.
+The default and best supported layout is `dot`.
+
+For example:
+```
+https://localhost:8080/debug/graph?layout=circo
+```
+
+Available layout options include:
+- `dot` (default) - Hierarchical layout
+- `neato` - Spring model layout
+- `fdp` - Force-directed layout
+- `sfdp` - Multiscale version of fdp for large graphs
+- `circo` - Circular layout
+- `twopi` - Radial layout
+- `osage` - Array-based layout
+- `text` - raw [DOT](https://graphviz.org/doc/info/lang.html) text of the graphfor use with 3rd-party service or tool
```

Comment:
```suggestion
- `dot` (default): Hierarchical layout
- `neato`: Spring model layout
- `fdp`: Force-directed layout
- `sfdp`: Multiscale version of fdp for large graphs
- `circo`: Circular layout
- `twopi`: Radial layout
- `osage`: Array-based layout
- `text`: raw [DOT](https://graphviz.org/doc/info/lang.html) text of the graph, for use with external tools
```
[s] prefer colons, fix typo, say 'external' instead of '3rd-party'